src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1023,7 +1003,16 @@ internal void GetTimePrecise(out int hour, out int minute, out int second, out i
         // Monday, 2 indicates Tuesday, 3 indicates Wednesday, 4 indicates
         // Thursday, 5 indicates Friday, and 6 indicates Saturday.
         //
-        public DayOfWeek DayOfWeek => (DayOfWeek)((InternalTicks / TicksPerDay + 1) % 7);
+        public DayOfWeek DayOfWeek
+        {
+            get
+            {
+                uint days = (uint)(UTicks / TicksPerDay) + 1;
+                if (!Bmi2.IsSupported) return (DayOfWeek)(days % 7);
+                // 613566757 = compute_unsigned_magic_info(7, 22) from https://github.com/ridiculousfish/libdivide/blob/master/doc/divide_by_constants_codegen_reference.c
```

Comment:
This needs mention in 3rd party notices.

Reply:
This is using just a number, not the any code?

Reply:
We prefer to err on giving more credit than necessary. Also, it makes the IP scans easier.

We do have https://github.com/lemire/fastmod referenced in the 3rd party notices already. Does it generate the same result?

Also, isn't the magic div/mod already handled by the JIT?

Reply:
https://github.com/dotnet/runtime/blob/703145649baf152e3678825be1849f6fd1bd2b24/src/coreclr/src/jit/lower.cpp#L5067